### PR TITLE
fix(table): Prevent duplicate scrollbars in certain browsers

### PR DESCRIPTION
**Pull request by macandcheese**
**Related Issue:** #8947 

## Summary
Fixes an issue in Chrome and Safari where double scrollbars would appear. As mentioned in the issue, it does seem to have to do with the conditional rendering of the assistive technology text. Firefox was unaffected. Honestly kind of a strange bug, I don't think (?) this has come up before with the AT hidden class we use. 

cc @geospatialem to verify this still works as expected in JAWS / NVDA - the issue was introduced with the recent addition of the new `interactionMode` property, so let's verify this doesn't have a negative effect on AT.

### CODE REVIEW

**Review by geospatialem — COMMENTED** [review]
Testing-wise looks good in Chrome on Windows and AT is still working as-intended with JAWS and NVDA. 👍🏻

**Review by driskull — APPROVED** [review]
🎉

File: packages/calcite-components/src/components/table-cell/table-cell.tsx [hunk 1]
```diff
@@ -254,18 +243,16 @@ export class TableCell
...
-              {(this.selectionCell || this.readCellContentsToAT) && (
-                <Fragment>
-                  {this.selectionCell && this.selectionText}
-                  {this.readCellContentsToAT && !this.selectionCell && this.contentsText}
-                </Fragment>
-              )}
-            </span>
+            {(this.selectionCell || this.readCellContentsToAT) && (
+              <span
+                aria-hidden={true}
```
[the remark this answers is not in the document]
Comment (macandcheese) [reply packages/calcite-components/src/components/table-cell/table-cell.tsx:248]: Sorry @jcfranco , for some reason this comment just showed up / sent an email notification. Can you open an issue to update and I can fix as this is already merged.

[the remark this answers is not in the document]
Comment (jcfranco) [reply packages/calcite-components/src/components/table-cell/table-cell.tsx:248]: ☝️ https://github.com/Esri/calcite-design-system/issues/8976

**Pull request merged**